lib/gc/ExecutionEngine/CPURuntime/Microkernel/BrgemmOnednn.cpp

```diff
@@ -57,6 +57,53 @@ static std::vector<brgemm_desc_t> g_brgemm_desc_list;
 static std::vector<brgemm_kernel_t *> g_brgemm_kernel_list;
 static std::vector<std::unique_ptr<char[]>> g_brgemm_palette;
 
+struct brgemm_cache_info_t {
+  std::shared_ptr<brgemm_desc_t> desc;
+  std::shared_ptr<brgemm_kernel_t> kernel;
+  std::shared_ptr<char> palette;
+
+  brgemm_cache_info_t() = default;
+  brgemm_cache_info_t(brgemm_desc_t *d, brgemm_kernel_t *k, char *p)
```

Comment:
ideally we need to change the `unique_ptr` in global palette pool to `shared_ptr` as well, and pass the `shared_ptr` of palette here for construction

Reply:
fixed.